x/wasm/keeper/events.go

```diff
@@ -65,3 +65,39 @@ func contractSDKEventAttributes(customAttributes []wasmvmtypes.EventAttribute, c
 	}
 	return attrs, nil
 }
+
+func newCallablePointEvent(customAttributes []wasmvmtypes.EventAttribute, contractAddr sdk.AccAddress, callstack []byte) (sdk.Events, error) {
+	attrs, err := callablePointSDKEventAttributes(customAttributes, contractAddr, callstack)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return sdk.Events{sdk.NewEvent(types.CallablePointEventType, attrs...)}, nil
+}
+
+func newCustomCallablePointEvents(evts wasmvmtypes.Events, contractAddr sdk.AccAddress, callstack []byte) (sdk.Events, error) {
+	events := make(sdk.Events, 0, len(evts))
+	for _, e := range evts {
+		typ := strings.TrimSpace(e.Type)
+		if len(typ) <= eventTypeMinLength {
+			return nil, sdkerrors.Wrap(types.ErrInvalidEvent, fmt.Sprintf("Event type too short: '%s'", typ))
+		}
+		attributes, err := callablePointSDKEventAttributes(e.Attributes, contractAddr, callstack)
+		if err != nil {
+			return nil, err
+		}
+		events = append(events, sdk.NewEvent(fmt.Sprintf("%s%s", types.CustomCallablePointEventPrefix, typ), attributes...))
+	}
+	return events, nil
+}
+
+func callablePointSDKEventAttributes(customAttributes []wasmvmtypes.EventAttribute, contractAddr sdk.AccAddress, callstack []byte) ([]sdk.Attribute, error) {
+	attrs, err := contractSDKEventAttributes(customAttributes, contractAddr)
+	if err != nil {
+		return nil, err
+	}
+	// attrs[0] is addr
+	attrs = append([]sdk.Attribute{attrs[0], sdk.NewAttribute(types.AttributeKeyCallstack, string(callstack))}, attrs[1:]...)
```

Comment:
Is there a reason to put callstack first? It's kind of weird.

Reply:
@loloicci Could you please answer this question?

Reply:
I did not care when I added this function. If it is better the call stack is in the tail of the attributes, I will leave it in another issue.

Reply:
Please check it and then if it is necessary, please add another issue, @loloicci .

Reply:
I left issue https://github.com/Finschia/wasmd/issues/45.